ERCS/erc-7779.md

```diff
@@ -0,0 +1,176 @@
+
+The storage variable would be using the `bytes32[]` type and is an append-only variable which newly delegated accounts append its storage slot hash. The account may choose to not append its storage base to an array if there is an identical entry that already exist.
+
+In case the account identifies that there are colliding storage bases, the account can perform further storage verification either off-chain or on-chain and decide whether the delegation would happen.
+
+Wallet may revert the delegation if the colliding storage includes a suspicious storage values that may target the user(e.g., shadow signer, etc).
+
+### `onRedelegation()`
+
+This function is to prepare for the redelegation to a new account.
+
+When this function is called, the existing delegated account should perform actions to not limit or impact the user when the user redelegates to a new account as much as possible.
+
+For example, the account could uninitialize the storage variables as much as it can to provide clean storage for new wallet.
+
+This standard, however, does not explicitly state the behavior to be done during this function call as wallet implementations have very distinctive architecture and details.
+
+**The standard expect the wallet implementation to revert it back to a clean storage state as much as possible with this function.**
+
+`onRedelegation()` should validate if the caller is indeed the authorized user by checking the `msg.sender` value.
+
+This could also be done through a `self-call` if a custom validation scheme is implemented or at the wallet's discretion as a side case.
+
+![diagram showing the flow of onredelegation](../assets/eip-7779/diagram.png)
+
+## Rationale
+
+EOA (Externally Owned Account) is currently comprised of cryptographic key pair that is mostly managed in the form of mnemonic phrase.
+
+This simplicity provided frictionless interoperability between wallets that gave users the freedom to freely migrate between different wallet applications.
+
+However, after the merge of [EIP-7702](./eip-7702.md), each EOA will be given the ability to delegate itself to a smart account which will impact migration as storage remains in the continuous context while EOA can be delegated to diverse smart accounts if the user migrates their wallet.
+
+Account Abstraction Wallets, given the wallet-specific validation and execution logic, also have the interoperability issue to be considered but its importance in EOA is much more significant as EOA users are already familiar with wallet migration and its a common action to migrate wallets.
+
+This spec provides a standard approach for fetching the storage base used in the delegated account together with an optional mechanism to clean up the storage.
```

Comment:
This content is justifying the proposal as a whole, and so belongs in the Motivation section. The Rationale section, on the other hand, should be used to explain choices made within the document. My favourite analogy is:

> Motivation: We have chosen to build a shed because...
> Rationale: We painted the shed red because...

Reply:
Thanks for the good feedback, I updated both the Motivation and the Rationale section.